--- arch/sh/kernel/setup.c.orig	2009-03-23 17:32:39.000000000 +0100
+++ arch/sh/kernel/setup.c	2009-03-23 17:33:51.000000000 +0100
@@ -185,10 +185,11 @@
 
 #ifdef CONFIG_BLK_DEV_INITRD
 	if (LOADER_TYPE && INITRD_START) {
-		/* INITRD_START is the offset from the start of RAM */
-
-		unsigned long initrd_start_phys = INITRD_START;
-		initrd_start_phys += __MEMORY_START;
+		/* STSDK - FAE/STAPIREF_COMPAT - Fix to use u-boot */
+                /* INITRD_START was originally the offset from the start of RAM */
+                /* but is redefined to be the offset from a page below the zero page */
+		unsigned long initrd_start_phys = 
+                       virt_to_phys(INITRD_START + PARAM - PAGE_SIZE);
 
 		if (initrd_start_phys + INITRD_SIZE <= PFN_PHYS(max_low_pfn)) {
 			reserve_bootmem(initrd_start_phys, INITRD_SIZE);
